pulp_container/app/registry_api.py

```diff
@@ -682,23 +706,30 @@ def put(self, request, path, pk=None):
                 task.delete()
                 raise Exception(str(error))
 
-        chunks = UploadChunk.objects.filter(upload=upload).order_by("offset")
-
-        with NamedTemporaryFile("ab") as temp_file:
-            for chunk in chunks:
-                temp_file.write(chunk.file.read())
-                chunk.file.close()
-            temp_file.flush()
-
-            uploaded_file = PulpTemporaryUploadedFile.from_file(File(open(temp_file.name, "rb")))
-
-        if uploaded_file.hashers["sha256"].hexdigest() == digest[len("sha256:") :]:
+        if artifact := upload.artifact:
+            artifact.touch()
```

Comment:
I think in this case we also need to check the sha256 for this artifact against digest.

Reply:
yes this a good check, let me add it.
should i raise same exception like in the codepath below? `raise Exception("The digest did not match")`?